source/adapters/cuda/image.cpp

```diff
@@ -1039,15 +1040,29 @@ UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesImportOpaqueFDExp(
     void *pNext = const_cast<void *>(pInteropMemDesc->pNext);
     while (pNext != nullptr) {
       const ur_base_desc_t *BaseDesc =
-          reinterpret_cast<const ur_base_desc_t *>(pNext);
+          static_cast<const ur_base_desc_t *>(pNext);
       if (BaseDesc->stype == UR_STRUCTURE_TYPE_EXP_FILE_DESCRIPTOR) {
-        const ur_exp_file_descriptor_t *FileDescriptor =
-            reinterpret_cast<const ur_exp_file_descriptor_t *>(pNext);
+        auto FileDescriptor =
+            static_cast<const ur_exp_file_descriptor_t *>(pNext);
 
         extMemDesc.handle.fd = FileDescriptor->fd;
         extMemDesc.type = CU_EXTERNAL_MEMORY_HANDLE_TYPE_OPAQUE_FD;
       } else if (BaseDesc->stype == UR_STRUCTURE_TYPE_EXP_WIN32_HANDLE) {
-        return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+        auto Win32Handle = static_cast<const ur_exp_win32_handle_t *>(pNext);
+
+        switch (memHandleType) {
+        case UR_EXP_EXTERNAL_MEM_TYPE_WIN32_NT:
+          extMemDesc.type = CU_EXTERNAL_MEMORY_HANDLE_TYPE_OPAQUE_WIN32;
+          break;
+        case UR_EXP_EXTERNAL_MEM_TYPE_WIN32_NT_DX12_RESOURCE:
+          extMemDesc.type = CU_EXTERNAL_MEMORY_HANDLE_TYPE_D3D12_RESOURCE;
+          extMemDesc.flags = CUDA_EXTERNAL_MEMORY_DEDICATED;
+          break;
+        case UR_EXP_EXTERNAL_MEM_TYPE_OPAQUE_FD:
```

Comment:
shouldn't this map to `CU_EXTERNAL_MEMORY_HANDLE_TYPE_OPAQUE_FD` instead of an error?

Reply:
I'm wondering if you've added all the cases you want, just because cuda has a fair few more?